File: FDS/Utilities/Matlab/scripts/ashrae_7.m
%Roger Wang
%7-20-11
%ashrae_7.m

close all
clear all

infile{1} = '../../Verification/HVAC/ashrae_7_exp.csv';
infile{2} = '../../Verification/HVAC/ashrae7_fixed_flow_devc.csv';
infile{3} = '../../Verification/HVAC/ashrae7_quadratic_devc.csv';
infile{4} = '../../Verification/HVAC/ashrae7_table_devc.csv';

label{1} = 'Experiment';
label{2} = 'Fixed Flow & ';
label{3} = 'Quadratic & ';
label{4} = 'Table & ';

duct{1} = '1';
duct{2} = '2';
duct{3} = '3';
duct{4} = '4';
duct{5} = '5';
duct{6} = '56';
duct{7} = '6';
duct{8} = '7';

M = csvread(infile{1},6,1);

for n = 2:4
    if ~exist(infile{n})
        display(['Error: File ',infile{n},' does not exist. Skipping case.'])
        return
    end
    m = csvread(infile{n},2,1);
    pressure(n,1:8) = m(length(m(:,1)),1:8);
end

filename = '../../Manuals/FDS_Verification_Guide/SCRIPT_FIGURES/ashrae_7.tex';
fid = fopen(filename, 'wt');

fprintf(fid,'%s\n','\begin{center}');
fprintf(fid,'%s\n','\begin{tabular}{|c|c|c|c|c|c|c|c|c|} \hline');
fprintf(fid, '%s', 'Duct Number & ');
fprintf(fid, '%s\n', '1 & 2 & 3 & 4 & 5 & 56 & 6 & 7 \\ \hline');
fprintf(fid, '%s\n', 'Experiment &');                                                                       
fprintf(fid, '%6.3f & %6.3f & %6.3f & %6.3f & %6.3f & %6.3f & %6.3f & %6.3f %s\n', M,'\\');
for i = 2:4
    fprintf(fid, '%s', label{i});
    fprintf(fid, '%6.3f & %6.3f & %6.3f & %6.3f & %6.3f & %6.3f & %6.3f & %6.3f %s\n', pressure(i,1:8),'\\');
end
fprintf(fid,'%s\n','\hline');
fprintf(fid,'%s\n','\end{tabular}');
fprintf(fid,'%s\n','\end{center}');

    
    

File: Utilities/Python/scripts/ashrae_7.py
# Roger Wang
# 7-20-11
# ashrae_7.m
#
# Converted by Floyd

import os
import numpy as np
import pandas as pd

datadir =  '../../Verification/HVAC/'
filename = [
    'ashrae_7_exp.csv',
    'ashrae7_fixed_flow_devc.csv',
    'ashrae7_quadratic_devc.csv',
    'ashrae7_table_devc.csv'
]

label = [
    'Experiment &',
    'Fixed Flow & ',
    'Quadratic & ',
    'Table & '
]

# duct = ['1', '2', '3', '4', '5', '56', '6', '7']

pressure = np.zeros((4, 8))

skip_case = False

for i in range(len(filename)):
   name = datadir+filename[i]
   if not os.path.exists(name):
      skip_case = True
      print('Error: File ', filename[i], ' does not exist. Skipping case.')

if skip_case: quit()

M = pd.read_csv(datadir+filename[0],skiprows=6,header=None)

for n in range(1, len(filename)):
   df_sim = pd.read_csv(datadir+filename[n],skiprows=2,header=None)
   last_row_data = df_sim.iloc[-1].values
   pressure[n, :] = last_row_data[1:9]

# --- Write to LaTeX File ---

texname = '../../Manuals/FDS_Verification_Guide/SCRIPT_FIGURES/ashrae_7.tex'

with open(texname, 'w') as fid:

   # Boilerplate LaTeX environment opening
   fid.write('\\begin{center}\n')
   fid.write('\\begin{tabular}{|c|c|c|c|c|c|c|c|c|} \\hline\n')

   # Header Row
   fid.write('Duct Number & 1 & 2 & 3 & 4 & 5 & 56 & 6 & 7 \\\\ \\hline\n')

   # Experiment Row (M)
   format_str_exp =label[0] + ' %6.3f &' * 7 + ' %6.3f \\\\ \n'

   # The '*' operator unpacks the NumPy array 'M' into arguments for the format string.
   fid.write(format_str_exp % tuple(M.iloc[0,1:9]))

   # Simulation Rows (pressure)
   for i in range(1,  len(filename)):

      # Build the format string using the label prefix
      format_str_sim = label[i] + ('%6.3f & ' * 7) + '%6.3f \\\\ \n'

      # Unpack the pressure data and write the line
      fid.write(format_str_sim % tuple(pressure[i,:]))

   # Boilerplate LaTeX environment closing
   fid.write('\\hline\n')
   fid.write('\\end{tabular}\n')
   fid.write('\\end{center}\n')

print(f"\n? Successfully generated LaTeX table in {texname}")